toffy/rosetta_test.py

```diff
@@ -217,3 +218,58 @@ def test_replace_with_intensity_image(overwrite, folders):
                     file = os.path.join(top_level_dir, 'run_{}'.format(dir),
                                         'fov{}'.format(fov), 'chan1' + suffix)
                     assert os.path.exists(file)
+
+
+def test_create_rosetta_matrices():
+    # step 1: create rosetta matrix
+    # step 2: save as csv
+    with tempfile.TemporaryDirectory() as temp_dir:
+        # Pulling in channel names
+        test_channels = [23, 48, 56, 69, 71, 89, 113, 115, 117, 141, 142, 143, 144, 145, 146]
+        test_channels += [147, 148, 149, 150, 151, 152, 153, 154, 155, 156, 157, 158, 159, 160]
+        test_channels += [161, 162, 163, 164, 165, 166, 167, 168, 169, 170, 171, 172]
+        test_channels += [173, 174, 175, 176, 181, 197]
+
+        # Creating random matrix to test
+        rand_np_matrix = np.random.randint(1, 50, size=[47, 47])
+        random_matrix = pd.DataFrame(rand_np_matrix, index=test_channels, columns=test_channels)
+        output_path = os.path.join(temp_dir, 'Random_matrix.csv')
+        random_matrix.to_csv(output_path)
+
+        # Checking channels = None
+        multipliers = [3]
+        create_rosetta_matrices(output_path, temp_dir, multipliers)
+        rosetta_path = os.path.join(temp_dir, 'Rosetta_Titration%s.csv'
+                                    % (str(multipliers[0])))
+        # grabs output of create_rosetta_matrices
+        test_matrix = pd.read_csv(rosetta_path, index_col=0).astype(int)
+        validation = (test_matrix / multipliers[0])
+
+        # confirm all channels scaled by multiplier
+        assert np.array_equal(random_matrix, validation)
+
+        # Checking Specific Channels
+        multipliers = [4]
+        channels = [197]
+        create_rosetta_matrices(output_path, temp_dir, multipliers, channels)
+
+        rosetta_path_2 = os.path.join(temp_dir, 'Rosetta_Titration%s.csv'
+                                      % (str(multipliers[0])))
+        # grabs output of create_rosetta_matrices
+        test_matrix_2 = pd.read_csv(rosetta_path_2, index_col=0).astype(int)
+        # row index of input channel
+        test_index = test_channels.index(channels[0])
```

Comment:
Instead of checking that the specified row has been modified by the multiplier and looping over all of the other rows, you could divide the `test_matrix` by a vector that has 1 for every channel which isn't supposed to be modified, and the multiplier for the channels that are supposed to be modified. Then you could use the same check as above, making sure the validation is equal to the random_matrix.